### ISSUE TITLE
Workflows: Reject (request changes) in a second step doesn't send emails to users in previous task

### ISSUE THREAD

**Issue by vladox** [wagtail/wagtail#10196]
### Issue Summary

When in a multistep workflow, after the first step rejecting doesn't produce any emails.

### Steps to Reproduce

1. Create a workflow with at least two steps
2. Assign a **group of users** to one step and another different group to the second step. 
3. Start the workflow in a page configured to run the workflow first step using a user in the **first group**. 
4. Reject requesting changes using a user in the **second group**, an email stating the rejection is sent.
5. Resume workflow using a user in the **first group,** an email is sent about the workflow being resumed.
6. Do changes and approve workflow step using user in the **second group,** an email is sent about the step being approved and ready for moderation.
7. Reject the workflow again using a user in the **first group**,  no email is sent due to condition `requested_by != triggering_user` in /wagtail/admin/mail.py:310

Steps 4 and 5 can be skipped to reproduce the issue, they have been included to demonstrate that in a specific case emails are being sent in case of a rejection. 

The expected behaviour is that when a workflow step is rejected the users in the previous task get notified, currently the `get_recipient_users` method in /wagtail/admin/mail.py:306 doesn't seem to support it not even notifying the user that submitted the step for moderation.

This can be somehow alleviated by setting the requested_by user to the user resuming the action on the workflow resume action method in /wagtail/core/models/__init__.py:3552
But as currently programmed the method `get_recipient_users` never returns more than one user, so a notification to all users in the group is not possible. 

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes (Wagtail 4.1.1)

### Technical details

-   Python version: 3.9
-   Django version: 3.2.18
-   Wagtail version: 4.1.1
-   Browser version:  Chrome Version 110.0.5481.177 (Official Build) (x86_64)

**Comment by vladox** [comment]
@lb- Can you try to reproduce this issue in your end?

**Comment by floels** [comment]
Hi @vladox,

Indeed, looking at the `get_recipient_users` method of `WorkflowStateRejectionEmailNotifier`:

https://github.com/wagtail/wagtail/blob/dc48dcce8a0480bf09629a16a5a3583d2234788e/wagtail/admin/mail.py#L340-L352

it seems that the intention is to send an email only to the user who triggered the workflow.

Why would you expect that:

> when a workflow step is rejected the users in the previous task get notified

?

**Comment by vladox** [comment]
Hi @floels 
Consider the following workflow setup (I imagine it should be a fairly common one), you have a multilanguage site and for the translated branch of your site (Wagtail-Localize) a workflow including two steps is setup:

1. Submit to translators for translation/correction
2. Approval by workflow initiator (editor/moderator) 

Now as currently coded if the translator rejects a change (maybe the text is missing or something like that) the workflow initiator gets a notification no issues.
Both on the contrary if in the step 2 the moderator rejects changes (much more likely) no email notification is sent to the translator warning them about some issue with the translation.

Do you see the issue? Is not the workflow initiator that needs to be notified but the user that resumed or submited the previous step.

Please let me know your thoughts about this use case.  
-

**Comment by jacobtoppm** [comment]
Yeah, this is not a bug, this is a feature request. Workflow will always notify the submitter to the workflow, but won't notify previous tasks - depending on use case that could cause a huge amount of email spam, since the previous tasks' reviewers will also get emails when the task is resubmitted to them, which to me is closer to when it's actually useful for them to be notified. That said, I see your use case...

For now, I'd suggest you implement this as a customisation using a https://docs.wagtail.org/en/stable/reference/signals.html#workflow-rejected signal handler.

If anyone else would benefit from this or a setting to add this, please let us know. I wouldn't be averse to us adding this, but if we do, I think it would benefit from an off switch - likely a setting.
